deepomatic/api/resources/task.py

```diff
@@ -83,13 +96,17 @@ def wait(self, timeout=60, wait_exp_multiplier=0.05, wait_exp_max=1.0):
 
     def _refresh_status(self):
         logger.debug("Refreshing Task {}".format(self))
-        self.refresh()
+        warn_on_http_retry_error(self.refresh, suffix="Retrying until Task.wait timeouts.")
         return self['status']
 
     def _refresh_tasks_status(self, pending_tasks, success_tasks, error_tasks, positions):
         logger.debug("Refreshing batch of Task {}".format(pending_tasks))
         task_ids = [task.pk for idx, task in pending_tasks]
-        refreshed_tasks = self.list(task_ids=task_ids)
+        functor = functools.partial(self.list, task_ids=task_ids)
+        refreshed_tasks = warn_on_http_retry_error(functor, suffix="Retrying until Task.batch_wait timeouts.")
```

Comment:
other errors still break the task wait loop if I understand correctly; do we want that?

Reply:
we do want that: other exceptions must not be retried because they would not update the task status.